[Estuary] New Stream selection dialog + related changes/fix

jjd-uk opened the pull request:
## Description
PR https://github.com/xbmc/xbmc/pull/25762 introduced the ability to access the stream selections outside of the video/audio/subtitles settings dialogs, and included Estuary changes to help with testing. This is a follow up to create a new Stream Selection dialog that can be access via a button on the OSD, replacing the exisitng Subtitle button.

Also included is a fix and some cleanup & other related changes.

Superceeds  https://github.com/xbmc/xbmc/pull/22557

## New

What was included in PR https://github.com/xbmc/xbmc/pull/25762

![image](https://github.com/user-attachments/assets/40e827e8-f824-4ceb-b28e-ea79a56aa2cb)

New Stream selection dialog

![image](https://github.com/user-attachments/assets/2f830ea6-0a78-43e5-bd4e-2a0a59bec8e3)

The icon for the Stream selection button was taken from Googles open source Material Design icon library

https://fonts.google.com/icons?selected=Material+Symbols+Outlined:playlist_add_check

## Changes

`SettingsDialogOSDContent` renamed to `SettingsDialogVideoSettingsContent` to better reflect what it opens.
`SettingsDialogMusicOSDContent` renamed to `SettingsDialogMusicSettingsContent` to better reflect what it opens and be consistent to video osd.

## Fix

The dialog select layout wasn't optimised for multiple lines

![image](https://github.com/user-attachments/assets/a0a5d65e-6b28-482a-b519-580cbadaea48)

Corrected layout

![image](https://github.com/user-attachments/assets/370ee03a-327a-471f-ac11-2e823fd885e3)

## How has this been tested?
Tested on Windows 10 with a variety of video files.

## Types of change
<!--- What type of change does your code introduce? Put an `x` with no space in all the boxes that apply like this: [X] -->
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [x] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` with no space in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

jjd-uk committed:
[Estuary] Stream selection changes

jjd-uk commented:
@CastagnaIT Thanks for the work to allow this to be done.

@mjibe this superceeds your https://github.com/xbmc/xbmc/pull/22557

Hitcher commented:
> @Hitcher in case sarbes is not avaiable, would you happen to understand what needs to be check for new textures.

I think this correct:

![streams](https://github.com/user-attachments/assets/c6339f46-a1d1-4b87-a6ec-227fc08dda64)

CastagnaIT commented:
> The dialog select layout wasn't optimised for multiple lines
![immagine](https://github.com/user-attachments/assets/9674d8b4-115c-4320-94e0-dd71656cf080)

imo its not so important give so importance to it to even enlarging the height of the layout,
probably just need to enable scrolling and remove carriage returns chars from source code,
text on the video tracks is quite rare,
and on adaptive streaming there may be high number of video tracks (10, 20 or also more) by bitrate, resolution and codecs,
so widen the item layout height becomes inconvenient for scrolling many items

I had already noticed this in the audio track dialog that there are already three lines and it's a little bit impractical to scroll,
I would like to try remove one line to reduce in to total 2 lines (merge 2 last lines in some way)

jjd-uk commented:
I forgot about https://github.com/xbmc/xbmc/pull/25762#discussion_r1830236369

Will consildate those variables and push an update.

jjd-uk commented:
> imo its not so important give so importance to it to even enlarging the height of the layout, 

The heights were all wrong anyway and needed to be changed as the bottom item would have it's height cut.

mjibe commented:
Hi @jjd-uk,

Thanks!

I wonder what to do to disable/enable subtitles ?

Wouldn't it be more convenient to add the toggle on your osd window?

I did for mine (in my PR), because sometimes I have to enable the (french) subtitles for my wife when she join me. Otherwise I'm setting them off and on given the context (when subs are wrongfully tagged "not forced" in mkv files).

IMO, it would be useful to add this instead of going inside submenus just to do this.

CastagnaIT commented:
> I used this sample for testing the multiple video tracks

you have to use this video sample to understand the situation for video case:
[DASH_multi_codec.strm.txt](https://github.com/user-attachments/files/17682049/DASH_multi_codec.strm.txt)
note that you need to install inputstream adaptive addon to play this video strm

> The heights were all wrong anyway and needed to be changed as the bottom item would have it's height cut.

Yes, but you are changing it with a very high item height,
where 90% of the time remains always empty because has no text
this fix as is, is counterproductive, will cause an unnecessary waste of space and ugly scrolling try to see with my sample

otherwise you should have a solution to make an item height based on the available lines of text

jjd-uk commented:
I could do a scroll to give something like

![image](https://github.com/user-attachments/assets/7374bf53-651f-4309-aeed-68d8c3919bc2)

jjd-uk commented:
> IMO, it would be useful to add this instead of going inside submenus just to do this.

@mjibe In case you missed it, you can now disable within the selection dialog

![image](https://github.com/user-attachments/assets/485c9f13-e11a-4467-9766-0c6eaa0b7b92)

CastagnaIT commented:
yeah for me this new compact version is very good,
if you can please try compact a bit also item layout on audio and subs dialogs

mjibe commented:
> > IMO, it would be useful to add this instead of going inside submenus just to do this.
> 
> @mjibe In case you missed it, you can now disable within the selection dialog
> 
> ![image](https://github.com/user-attachments/assets/485c9f13-e11a-4467-9766-0c6eaa0b7b92)
> 
> 
> 

My bad, talked too fast 😅 my comment was based on the screenshots in this PR, as I'm currently off my PC, didn't pull this PR to check it live.

Sorry again!

jjd-uk commented:
@CastagnaIT 

Tweaked the heights, not pushed yet. If it looks better for you then I'll push to PR.

![image](https://github.com/user-attachments/assets/f6a097a1-fd79-4c88-897b-b2ba15d2ba69)

![image](https://github.com/user-attachments/assets/7d386077-b763-4f38-8974-25dcf7040181)

CastagnaIT commented:
for me its good, just align the tick icon with baseline of first text line

jjd-uk commented:
Updates pushed

![image](https://github.com/user-attachments/assets/7e3affff-d98f-437d-b481-1e29dea2f6e2)

![image](https://github.com/user-attachments/assets/e54dc286-8b62-4a2e-8e64-4ee3b27e7f61)

![image](https://github.com/user-attachments/assets/06ea6141-43f4-4f0a-bc54-68fa08cea1ba)

KarellenX commented:
@jjd-uk If my streams are named, eg "Directors Commentary", Forced, SDH, Audio Described, will that also be visible in the listings? I don't see any of that in your screenshots, and many times I have multiple English streams, so need to know which is which.

CastagnaIT commented:
seem very good, i will test tomorrow or sunday

jjd-uk commented:
> @jjd-uk If my streams are named, eg "Directors Commentary", Forced, SDH, Audio Described, will that also be visible in the listings? I don't see any of that in your screenshots, and many times I have multiple English streams, so need to know which is which.

If the stream has a user added description then this will be shown, then the standard flags are also shown.

![image](https://github.com/user-attachments/assets/9309bec2-d73c-4f31-b718-cfb47c502e6c)

KarellenX commented:
Thanks @jjd-uk

jjd-uk commented:
FYI the standard flags added by @CastagnaIT are:

default
forced
captions
audio description
original

These are shown on Line 1.
Then user description is Line 2.

jjd-uk commented:
Something I forgot to point out in PR description.

I've made the Stream selection button and dialog always available, even if there's no multiple subs, audio and video, that way it's always avaiable to quickly see the stream properties. Instead the buttons so opening the stream selection are either enabled or disabled via conditions, so for example if there's only a single audio stream the button is disabled, however you can still see the limited properties given.

CrystalP commented:
This will be very handy!

The checkmark seems redundant to me, it's not there in other dialogs, where the default selection + yellow selected item color are enough to know which is the selected item.

For some reason the audio codec type doesn't show for me in the audio selection for some videos but it's fine in stream selection and master is fine:
![image](https://github.com/user-attachments/assets/21598bc0-7cde-4f18-9504-536e03c8e4ef)

![image](https://github.com/user-attachments/assets/6ad251af-28f9-4a62-aaa7-059e93932ba1)

There is an inconsistency in the treatment of unknown language: appears blank in stream selection but says Unknown as expected in subtitle/audio selection:
![image](https://github.com/user-attachments/assets/21598bc0-7cde-4f18-9504-536e03c8e4ef)

![image](https://github.com/user-attachments/assets/6bc29abc-2661-4857-9f3e-432feb8945a4)
It seems there is always a text in other situations (none, disabled, the three letter language code).

Shouldn't "Stream selection" be "Streams selection" (plural)?

CastagnaIT commented:
> The checkmark seems redundant to me, it's not there in other dialogs, where the default selection + yellow selected item color are enough to know which is the selected item.

it could, but I find the tick mark is more useful than having a different color text, considering that tick icon can be helpful to people with disabilities

jjd-uk commented:
> Shouldn't "Stream selection" be "Streams selection" (plural)?

You can only playback a single stream of each type which is why I went with the singular rather than plural. So for me as a native speaker Stream seemed more correct.

jjd-uk commented:
There is some inconsistencies between what I display in the label 2 for the button in the Stream selection dialog, and in the actual selection dialog, as different infolabels have to be used.

jjd-uk commented:
@CrystalP Fix ups done.

For the Subtitles `Unknown` the infolabel `$INFO[VideoPlayer.SubtitlesLanguage]` doesn't return anything, so I adjusted `VideoStreamDialogItemLabelVar` to account for this.

In addition I noticed a movie in my collection where audio language was `Unknown` so have made adjustments for this also.

![image](https://github.com/user-attachments/assets/043865c1-019f-4739-8c6a-bf41d037be50)

So now dialog shows:

![image](https://github.com/user-attachments/assets/2caf5cbf-32ce-49b2-960d-4349a357d6f4)

For your audio codec problem, I assume `$INFO[ListItem.Property(stream.codecdesc)]` does not return anything. So I've moved `$INFO[ListItem.Property(stream.codecdesc)]` along side audio language then added `$INFO[ListItem.Property(stream.codec)]` on second line which should ensure audio codec is always shown.

![image](https://github.com/user-attachments/assets/b726245f-a465-4ccf-ba2f-c11a64ee2c52)

jjd-uk commented:
Although I'm not sure we need so much dupplication for the audio codec so perhaps something like

```
	<variable name="StreamSelectionAudioCodecVar">
		<value condition="String.IsEmpty(ListItem.Property(stream.codecdesc))">$INFO[ListItem.Property(stream.codec)]$INFO[ListItem.Property(stream.channels), $LOCALIZE[31612]: ]</value>
		<value>$INFO[ListItem.Property(stream.codecdesc)]</value>
	</variable>
```
would be better giving

![image](https://github.com/user-attachments/assets/33bc158b-65b6-444b-8014-5ae994fb0484)

But I don't appear to have anything that doesn't return `ListItem.Property(stream.codecdesc)`

CastagnaIT commented:
> Although I'm not sure we need so much dupplication for the audio codec so perhaps something like
> 
> ```
> 	<variable name="StreamSelectionAudioCodecVar">
> 		<value condition="String.IsEmpty(ListItem.Property(stream.codecdesc))">$INFO[ListItem.Property(stream.codec)]$INFO[ListItem.Property(stream.channels), $LOCALIZE[31612]: ]</value>
> 		<value>$INFO[ListItem.Property(stream.codecdesc)]</value>
> 	</variable>
> ```
> 

"codecdesc" has nothing to do with "codec+channels" strings,
so the first line (codec+channels) must be always shown,
"codecdesc" its only an optional description, it can contain any text that does not concern the codec name and channels, you can't use it as "condition"

jjd-uk commented:
@CastagnaIT In that screenshot:

`$INFO[ListItem.Property(stream.codecdesc)] `returns the `AC3 5.1(side) `and `TrueHD 7.1`

`$INFO[ListItem.Property(stream.codec)]$INFO[ListItem.Property(stream.channels), $LOCALIZE[31612]: ]` returns `ac3 channels: 6` and `truehd channels: 8`

In  your PR you only used `$INFO[ListItem.Property(stream.codecdesc)]` however from what @CrystalP posted this is not always filled, so I added `$INFO[ListItem.Property(stream.codec)]` which is always filled as far as I can see.

For the $INFO[ListItem.Property(stream.codecdesc)] returning the` `AC3 5.1(side)`` and `TrueHD 7.1` looks to me to be a more user friendly format, but it seems we can't rely on it alone.

It is [ListItem.Property(stream.description) that returns description of stream if set as shown here

![image](https://github.com/user-attachments/assets/864ac49a-12c3-4e96-b07a-3a2e2d13e428)

Regarding my tests, yes currently I'm only able to test with local files from my nas (which are a mixture of different types) as I do not inputstream adaptive set up currently.

jjd-uk commented:
Ok, installed inputstream.adaptive and tried that `DASH_multi_codec.strm`  to see if there's any difference from my nas stored files, but that only gives me multiple video streams.

![image](https://github.com/user-attachments/assets/f10089f6-f06d-4812-a3f6-d3670150ad3b)

CrystalP commented:
I pointed out the checkmark for style consistency, not disagreeing that it is more noticeable.

The "empty" codec desc case is the playback of a DVD through the menu. With the last changes the info is now duplicated lol:
![image](https://github.com/user-attachments/assets/aa08c2d3-63d3-4f5d-a3f5-a149a713915f)

jjd-uk commented:
OK I've rolled back the changes I did for the Audio Stream selection dialog. As it sounds like if ListItem.Property(stream.codecdesc) doesn't return anything that should be investigated.

So labels are back to being:

`<label>$INFO[ListItem.Label]$VAR[VideoStreamDialogItemLabelVar, - [LIGHT],[/LIGHT]]</label>`
`<label>$INFO[ListItem.Property(stream.codecdesc),, - ]$INFO[ListItem.Property(stream.channels), $LOCALIZE[31612] ]$INFO[ListItem.Property(stream.description), - ]</label>`

![image](https://github.com/user-attachments/assets/ef8300d8-e91e-4fb1-8d6a-356124962530)

jjd-uk commented:
On `ListItem.Property(stream.codecdesc)` issue, I've found a DVD iso file amongst my test files and that is also not returning anything, so maybe an issue with full disc rips???

![image](https://github.com/user-attachments/assets/acfbdce8-fb41-4178-9673-6ddc1435c1aa)

CrystalP commented:
I found the issue with the empty codedesc and it's in a path unique to DVD playback. This is a fix https://gist.github.com/CrystalP/994e9183766759a950d5ee7dad4a05c3, I can submit a PR.

jjd-uk commented:
> I found the issue with the empty codedesc and it's in a path unique to DVD playback. Here is a fix https://gist.github.com/CrystalP/994e9183766759a950d5ee7dad4a05c3, I can submit a PR.

As this PR is only for skin changes, please submit separately.

CastagnaIT commented on addons/skin.estuary/xml/Includes_DialogSelect.xml:
```diff
@@ -259,25 +259,27 @@
 						<aligny>center</aligny>
-						<label>$INFO[ListItem.Property(stream.codec)]$INFO[ListItem.Property(stream.resolution),$COMMA ]$INFO[ListItem.Property(stream.bitrate),$COMMA , kbps]$INFO[ListItem.Property(stream.fps),$COMMA , fps]$VAR[VideoStreamDialogVideoItemLabelVar, - [LIGHT],[/LIGHT]]</label>
+						<scroll>true</scroll>
+						<label>$INFO[ListItem.Property(stream.codec)]$INFO[ListItem.Property(stream.hdrtype, - )]$INFO[ListItem.Property(stream.resolution), - ]$INFO[ListItem.Property(stream.fps), - , fps]$VAR[VideoStreamDialogItemLabelVar, - [LIGHT],[/LIGHT]]</label>
```
please restore the bitrate string where it was, without it its impossible to distinguish video tracks
![immagine](https://github.com/user-attachments/assets/2bc9000d-04d5-42fd-833c-fd17fbe15738)

jjd-uk replied:
I removed it because everything I had displayed 0 kbps, and I didn't think you would have a choice where bitrate would be only difference. As you've shown this to be wrong I will look to restore bitrate, but see if I can do a check for a 0 value.

CastagnaIT replied:
oh right i had completely forgotten about the 0 yes a solution need to be found,
if not from skin side, from code by change the bitrate type from int to string value, and so when 0 return empty string

jjd-uk replied:
New variable added so a conditional check for 0 can be done.
```
	<variable name="StreamBitrateVar">
		<value condition="!String.IsEqual(ListItem.Property(stream.bitrate),0)">$INFO[ListItem.Property(stream.bitrate), - , kbps]</value>
	</variable>
```

CastagnaIT commented on addons/skin.estuary/xml/Includes_DialogSelect.xml:
```diff
@@ -440,25 +443,27 @@
+						<font>font10</font>
+						<aligny>center</aligny>
+						<scroll>true</scroll>
+						<label>$INFO[ListItem.Property(stream.codecdesc),, - ]$INFO[ListItem.Property(stream.channels), $LOCALIZE[31612] ]$INFO[ListItem.Property(stream.description), - ]</label>
```
there is a surplus space
```suggestion
<label>$INFO[ListItem.Property(stream.codecdesc),, - ]$INFO[ListItem.Property(stream.channels),$LOCALIZE[31612] ]$INFO[ListItem.Property(stream.description), - ]</label>
```

jjd-uk replied:
I will check, but I'm sure the space is needed before the Channels label to separate from  codedesc value. Do we even the Channels value? as channel layout always seems to be part of codedesc, unless you see cases where it's not present from addon streams.

jjd-uk replied:
Actually I probably do need to some adjustments for this since as I was trying to account for possibility of no codedesc value. However as CrystalP has found cause of no codedesc value for DVD image/folder files, then there's no need to account for this anymore.

CastagnaIT replied:
stream.codecdesc have suffix of " - " a space is already included should in theory be enough

stream.codecdesc could include a "channel" string value, but it come from ffmpeg metadata such as mkv,
with all others demuxers such as pvr/inpustreams etc... dont exists

I would have thought of removing this
https://github.com/xbmc/xbmc/blob/25915bd45c9f4f4cf97dddeb1c2b88923a47ffaa/xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemux.cpp#L129
so that the codec description stays clean

it would probably be more appropriate to have our own text conversion of the channels value instead of outsourcing the task to ffmpeg, so that can be used by all demuxers

jjd-uk replied:
Adjustments made

jjd-uk commented:
Update pushed.

- For Video streams restored steam.bitrate via a variable to allow for a check of a 0 value.

- For Audio stream made stream.channels match the label style of audio selection under Audio setting e.g. 6 channels.

Audio Settings

![image](https://github.com/user-attachments/assets/23c4baf8-0c1c-4fc4-bee1-9e7685c006b8)

Audio stream selection

![image](https://github.com/user-attachments/assets/79aa2cdb-831c-4c36-9f42-bd8c7db43637)

jjd-uk commented:
Regarding the display of the channels count.

In this

![image](https://github.com/user-attachments/assets/15ceb5fa-bddc-4872-bfbb-d00eb1ff1c95)

The `5.1(side)` is part of what is returned by `codedesc` so is the following `6 channels` really needed?

However for consistency with what is display via Audio settings I've kept the seperate channel count.

CastagnaIT commented:
> The `5.1(side)` is part of what is returned by `codedesc` so is the following `6 channels` really needed?
> 
> However for consistency with what is display via Audio settings I've kept the seperate channel count.

i answered you before read my comment above https://github.com/xbmc/xbmc/pull/25947#discussion_r1835660992
this should be changed and improved on the source code

jjd-uk commented:
I think the dialog width needs to be increased by about anorther 100 as just found something I cuold use to test full Video stream properties. At the moment it scrolls:

![image](https://github.com/user-attachments/assets/4a570e07-4081-4a30-8c27-f325cb8cc267)

An increase in width of 100 should prevent this.

CastagnaIT commented:
> I think the dialog width needs to be increased by about another 100 as just found something I could use to test full Video stream properties. At the moment it scrolls:
> 
> ![image](https://private-user-images.githubusercontent.com/5781142/384687849-4a570e07-4081-4a30-8c27-f325cb8cc267.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzEyNDE4NDQsIm5iZiI6MTczMTI0MTU0NCwicGF0aCI6Ii81NzgxMTQyLzM4NDY4Nzg0OS00YTU3MGUwNy00MDgxLTRhMzAtOGMyNy1mMzI1Y2I4Y2MyNjcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTBUMTIyNTQ0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZWYyNmUyMjU3MDY0NTQ5NWFjZmRmNjIwMDZmODkzNDhlNTVhYzhlY2VmZWQ0NjE4Njk1NWVlN2VhNGRkMTI3NyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.vcqVVsdunon2G9KkWRAj-uQZvgW9li9j0p2MQdO2EXs)
> 
> An increase in width of 100 should prevent this.

because you added hdr type, that occupy space,
also using " - " instead of ", " take much more space
my original solution was more clear
![immagine](https://github.com/user-attachments/assets/329c49ba-7b86-476b-bd5e-61e86c27d6ab)
and the bitrate has more sense to be on the left of the resolution

jjd-uk commented:
Yeah my personal preference is to use a dash as a separator, however you're right that a comma would be more compact so we can keep existing width.

![image](https://github.com/user-attachments/assets/06db4d2f-b1d0-4ff6-be82-bcd8d21e4e2d)

CastagnaIT commented:
seem good, anyway with higher bitrate text will inevitably become too long...otherwise the hdr should be placed to 2nd line

CastagnaIT commented:
maybe expand anyway a bit the width of dialog of some pixels to have some little additional space

CastagnaIT commented:
if you are able try to adjust space between external border
![immagine](https://github.com/user-attachments/assets/069ea856-f597-4980-9811-02e5b5d05c83)
![immagine](https://github.com/user-attachments/assets/6d30d393-b7fa-4522-9023-f415f08e8778)
the star, compared to the (left of) tick, have more pixel spaces on the right that can be reused for text, few pixels recovered but better than nothing

jjd-uk commented:
Expanding width by 50 gives

![image](https://github.com/user-attachments/assets/deb3eaa1-ee64-4f97-8f12-175bd3131462)

Note - my UHD rips only have a single video stream so I am disabling stream count check on local build to test.

jjd-uk commented:
The width change has now been pushed to PR.

CastagnaIT reviewed (approved):
i made some test, and lgtm

i tested also a DVD with the cristalp patch,
despite the patch works to add codecdesc, are needed other changes to remove rendundant info's from the dvd demuxer, otherwise channels string are shown doubled

if @CrystalP want include also the other changes on the dvd demuxer to remove rendundant info, otherwise i will make a PR that includes his changes as well

CrystalP commented:
Yes I noticed and removed the extra text in the name, see the submitted PR.

I need to try the latest version a bit more, but at least for subtitles, I'd prefer to have the track name concatenated on the first line.
It's important for anime to distinguish the tracks with signs & song, dialogue, dialogue with/without honorifics, subs by different groups, etc.
Not as important for audio tracks afaik but wouldn't hurt for consistency.

To remove the channels field: maybe soon after ffmpeg7 merge, not sure what it will take to show the atmos indication.

jjd-uk commented:
@CrystalP once you've had chance to test again, please provide screenshots of any issue you see with the subtitle types you talk about. Even better would be if you could provide a sample, ping me on Slack.

Hitcher commented:
There's not much padding around the icon compared to other buttons so I suggest reducing its size and repositioning it slightly.

Before:

![screenshot00023](https://github.com/user-attachments/assets/5dcd5197-4513-4c61-88b3-36f3c987427e)

After:

![screenshot00024](https://github.com/user-attachments/assets/91e09aaf-bcda-4c22-b1fb-bb474045a67f)

Here's my adjusted one if you want to use that; I've also converted it to greyscale:

![streams](https://github.com/user-attachments/assets/221ec60d-48ac-4f74-b9e2-1229db3aec72)

Hitcher commented:
Also noticed that the itemlayout labels are scrolling which looks awful when there's quite a few of them scrolling - only the focusedlayout labels should scroll.

Hitcher reviewed (approved):
Everything looks good, thanks.

jjd-uk commented on addons/skin.estuary/xml/Includes_DialogSelect.xml:
```diff
@@ -307,36 +307,38 @@
 						<scroll>true</scroll>
-						<font>font14</font>
-						<label>$INFO[ListItem.Property(stream.codec)]$INFO[ListItem.Property(stream.resolution),$COMMA ]$INFO[ListItem.Property(stream.bitrate),$COMMA , kbps]$INFO[ListItem.Property(stream.fps),$COMMA , fps]$VAR[VideoStreamDialogVideoItemLabelVar, - [LIGHT],[/LIGHT]]</label>
+						<label>$INFO[ListItem.Property(stream.codec)]$INFO[ListItem.Property(stream.hdrtype,$COMMA ]$INFO[ListItem.Property(stream.resolution),$COMMA ]$VAR[StreamBitrateVar,$COMMA , kbps]$INFO[ListItem.Property(stream.fps),$COMMA , fps]$VAR[VideoStreamDialogItemLabelVar, - [LIGHT],[/LIGHT]]</label>
```
I guess that crept in when I switched back to commas for the video properties. Fixed now.

sarbes commented on addons/skin.estuary/media/osd/fullscreen/buttons/streams.png:
The icon has issues in the "transparent" regions.

jjd-uk replied:
@sarbes  Then you'll have to tell us exactly what needs to be done as I have no clue and @hitcher said he followed your guidelines.

sarbes replied:
Just make the pixels in the color channel white. This is how it looks like with the alpha channel hidden:

![image](https://github.com/user-attachments/assets/862a1ea7-0f24-44cb-8115-9790cc0effb6)

sarbes replied:
The black area shows up as a grey outline: 
![image](https://github.com/user-attachments/assets/e915f95c-0673-4833-a5bf-e82e6efbb9c1)

jjd-uk replied:
@sarbes new icon supplied by hitcher pushed.
